builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -774,17 +774,38 @@ local function create_change_setting_formspec(dialogdata)
 	)
 end
 
+local function set_setting_if_changed(name, type, new_value)
+	if type == "bool" and new_value ~= core.settings:get_bool(name) then
+		print("Setting ["..name.."] changed. Invoking core.settings:set* ...")
+		core.settings:set_bool(name, new_value)
+		return true
+	elseif (type == "noise_params_2d" or type == "noise_params_3d") and
+			new_value ~= core.settings:get_np_group(name) then
+		print("Setting ["..name.."] changed. Invoking core.settings:set* ...")
+		core.settings:set_np_group(name, new_value)
+		return true
+	elseif new_value ~= core.settings:get(name) then
```

Comment:
If `type` is `"boolean"` and `new_value == core.settings:get_bool(name)`, this condition will always be true as `new_value` is a bool and `settings:get` always returns `nil` or a string, afaik.

Reply:
That's actually possible. I wondered why I didn't encounter this problem while testing, and then realised that double-clicking on boolean settings doesn't display the change setting dialog at all, and instead immediately toggles the value, lol.

That means testing for `type == bool` in this function and in `handle_change_setting_buttons` is redundant - the functions never get invoked for boolean setting changes. I'll remove the bool check in `set_setting_if_changed`, and wait for more opinions before removing that check from `handle_change_setting_buttons`.